backend/runner/plancheck/statement_report_executor_affected_rows.go

```diff
@@ -34,6 +37,43 @@ func getTableDataSize(metadata *storepb.DatabaseSchemaMetadata, schemaName, tabl
 	return 0
 }
 
+func buildGetTableDataSizeFunc(metadata *storepb.DatabaseSchemaMetadata) base.GetTableDataSizeFunc {
+	var schemaMetadata *model.DBSchema
+	return func(schemaName, tableName string) int64 {
+		// initialize schemaMetadata when the function is first used.
+		if schemaMetadata == nil {
+			schemaMetadata = model.NewDBSchema(metadata, nil /* schema */, nil /* schema */)
+		}
+		if schemaMetadata == nil {
+			return 0
+		}
+		dbMeta := schemaMetadata.GetDatabaseMetadata()
+		if dbMeta == nil {
+			return 0
+		}
+		schemaMeta := dbMeta.GetSchema(schemaName)
+		if schemaMeta == nil {
+			return 0
+		}
+		tableMeta := schemaMeta.GetTable(tableName)
+		if tableMeta == nil {
+			return 0
+		}
+		return tableMeta.GetRowCount()
+	}
+}
+
+func buildGetRowsCountByQuery(sqlDB *sql.DB, engine storepb.Engine) base.GetAffectedRowsCountByQueryFunc {
+	return func(ctx context.Context, statement string) (int64, error) {
+		switch engine {
+		case storepb.Engine_OCEANBASE:
+			return getAffectedRowsCount(ctx, sqlDB, fmt.Sprintf("EXPLAIN FORMAT=JSON %s", statement), getAffectedRowsCountForOceanBase)
+		default:
```

Comment:
Shouldn't this be `case storepb.Engine_MYSQL`?

Reply:
mysql and maridb?

Reply:
mariadb may (or may not, I don't know) return different results from EXPLAIN statements. So just mysql.